src/chat.rs

```diff
@@ -3598,6 +3598,49 @@ mod tests {
         Ok(())
     }
 
+    /// Test that group updates are robust to lost messages and eventual out of order arrival.
+    #[async_std::test]
+    async fn test_modify_chat_lost() -> Result<()> {
+        let alice = TestContext::new_alice().await;
+
+        let bob_id = Contact::create(&alice, "", "bob@example.net").await?;
+        let claire_id = Contact::create(&alice, "", "claire@foo.de").await?;
+        let daisy_id = Contact::create(&alice, "", "daisy@bar.de").await?;
+
+        let alice_chat_id = create_group_chat(&alice, ProtectionStatus::Unprotected, "foo").await?;
+        add_contact_to_chat(&alice, alice_chat_id, bob_id).await?;
+        add_contact_to_chat(&alice, alice_chat_id, claire_id).await?;
+        add_contact_to_chat(&alice, alice_chat_id, daisy_id).await?;
+
+        send_text_msg(&alice, alice_chat_id, "populate".to_string()).await?;
+        let add = alice.pop_sent_msg().await;
+        async_std::task::sleep(std::time::Duration::from_millis(1100)).await;
```

Comment:
sleeps in tests are and, eventually probably, flaky.  why is it needed?  can it be avoided?

Reply:
Same as in the test above, it is needed to avoid smeared timestamps going into the future, because otherwise Bob will rewind timestamp of both "remove member" messages with `min(timestamp, now)` and apply the message `remove1` when it is received, restoring 3 members in a group.